src/aura/CRLP_Rollup/CRLP_RollupHelper.js

```diff
@@ -63,278 +82,643 @@
         //we check to see if mode is null since the change handler is called when mode is cleared in the container
         var mode = cmp.get("v.mode");
         console.log(mode);
-        if(mode != null) {
+        if(mode !== null) {
             console.log("Mode is " + mode);
             console.log("In changeMode");
 
-            //for create we don't need to set anything yet since mode will be changed to clone before viewing page
-            if(mode != "create") {
-                //check operations to appropriately set "N/A" for date/amount field values
-                this.changeOperationsOptions(cmp);
-                this.changeYearlyOperationsOptions(cmp, false);
-
-                //set readonly fields if mode is View, else allow user to make changes
-                if (mode === "view") {
-                    cmp.set("v.isReadOnly", true);
-                } else {
-                    cmp.set("v.isReadOnly", false);
-                    this.resetSummaryObjects(cmp, cmp.get("v.activeRollup.Detail_Object__r.QualifiedApiName"));
-                    if (!$A.util.isEmpty(cmp.get("v.objectDetails"))) {
-                        this.resetAllFields(cmp);
+            //View is the only readOnly mode. Clone removes the activeRollupId for save.
+            //Create hides all fields and sets isIncomplete to disable the save button.
+            if (mode === "view") {
+                cmp.set("v.isReadOnly", true);
+            } else if (mode === "clone") {
+                cmp.set("v.activeRollupId", null);
+                cmp.set("v.isReadOnly", false);
+            } else if (mode === "edit") {
+                cmp.set("v.isReadOnly", false);
+            } else if (mode === "create") {
+                cmp.set("v.isIncomplete", true);
+                cmp.set("v.isReadOnly", false);
+                this.hideAllFields(cmp);
+            }
+        }
+    },
+
+    filterFieldsByType: function(cmp, typeList, allFields) {
```

Comment:
This applies to all the javascript, but I'm dropping the comment here. I'd like to have all of the JS methods documented the same way our apex methods are documented. That will make it much easier to support the code long term. For example:

```
/*
* @description: {describe what this function does}
* @param typeList {...}
* @param allFields {...}
*/
filterFieldsByType: function(cmp, typeList, allFields) {
```